src/Uno.UI/UI/Xaml/Controls/ProgressRing/ProgressRing.wasm.cs

```diff
@@ -1,22 +1,63 @@
 ﻿#if __WASM__
 using System;
-using System.Collections.Generic;
-using System.Text;
-using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Controls.Primitives;
 
 namespace Windows.UI.Xaml.Controls
 {
-    public partial class ProgressRing : Control
+	public partial class ProgressRing : Control
 	{
 		private static void OnForegroundChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
 		{
-			
+
 		}
 
 		private static void OnIsActiveChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
 		{
-		
+			var progressRing = dependencyObject as ProgressRing;
+			var isActive = args.NewValue as bool?;
+
+			if (progressRing != null && progressRing.IsLoaded && isActive.HasValue)
+			{
+				VisualStateManager.GoToState(progressRing, isActive.Value ? "Active" : "Inactive", false);
+			}
+		}
+
+		protected override void OnLoaded()
+		{
+			base.OnLoaded();
+			// The initial call to OnIsActiveChanged fires before ProgressRing is Loaded, so we also need to set a proper VisualState here
+			VisualStateManager.GoToState(this, IsActive ? "Active" : "Inactive", false);
 		}
+
+		public ProgressRingTemplateSettings TemplateSettings
+		{
+			get
+			{
+				var result = new ProgressRingTemplateSettings()
+				{
+					EllipseDiameter = 3,
+					MaxSideLength = 100
+				};
+
+				var size = Width.IsNaN() ? MinWidth : Width; // Strange, but ActualWidth is not working correctly here
+				result.EllipseOffset = new Thickness(size * (Math.Sqrt(2) - 1) / 2); // This is the difference between inscribed and circumscribed circle, it ensures that dots will be visible after control rectangle clipping
+
+				return result;
+			}
+		}
+	}
+}
+
+namespace Windows.UI.Xaml.Controls.Primitives
+{//TODO: I'm not an expert in TemplateSettings logic so it's just a stub to make standard ProgressRing UWP style bindings work in Wasm
+	public partial class ProgressRingTemplateSettings : DependencyObject
```

Comment:
Should be in a separate file

Reply:
Totally agree, I just didn't know if I had rights to create new files, GitHub workflows are still new to me.
Can I fix it in my fork and somehow append the changes to the current PR?

Reply:
Maybe I should append it to the existing [ProgressRingTemplateSettings.cs](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml.Controls.Primitives/ProgressRingTemplateSettings.cs) for other heads instead of creating a new file?

Reply:
Yes, you can create the file and commit it and push, and it'll be included in the PR (as you've done for the release notes).

The file you linked to is the generated stub, you should put the implementation in a different file.

Reply:
Done